Manager/src/main/java/io/raspberrywallet/manager/common/http/ApacheHttpClient.java

```diff
@@ -0,0 +1,58 @@
+package io.raspberrywallet.manager.common.http;
+
+import org.apache.commons.lang.NotImplementedException;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.fluent.Form;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicHeader;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ApacheHttpClient extends HttpClient {
```

Comment:
I don't get the naming idea, `ApacheHttpClient` extends `HttpClient`, where  `HttpClient` uses `org.apache.` libraries so it's already coupled with Apache, and cannot be reused in other ways.

Reply:
Isn't it good practice to add another level of abstraction? There will be only single point to change, when we would like to change apache to something else, since we use it through this class

Reply:
Sure, but then the most abstract class (HttpClient) should be independent from Apache. ApacheHttpClient should overlay this dependency

Reply:
Okay, I understand, but still can't see how to make it right, could you please give me an idea, how should I name these 3 classes?

Reply:
Sure, first of all I would ask a question if we are planing to use more than one httpclients, 
- if yes, then, I would move `JSONObject convert(Form body)` to some Utils class, because it's stateless and make this class dependent of `org.apache.http.client.fluent.Form`.
Then I would generify the class to:
```java
public abstract class HttpClient<Response, Body> {
    public abstract Response sendPOSTRequest(Body body, String endpoint) throws IOException;
    public abstract Response sendGETRequest(Body body, String endpoint);
}
```
- If not I would just rename `HttpClient` to `ApacheHttpClient`, `ApacheHttpClient` to `UnsecureHttpClient` and leave `SecureHttpClient` as is.

And after all, I would rebase onto http://square.github.io/okhttp/  😄